docs/DeveloperGuide.md

```diff
@@ -1,34 +1,503 @@
+### Introduction to *25HoursADay*
+*25HoursADay* is a scheduling app optimized for users via a convenient yet powerful Command Line Interface (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, *25HoursADay* can schedule your tasks and events faster than traditional GUI apps.
 
+### Purpose and scope
+The purpose of this developer guide is to describe the architecture and software design decisions for the task scheduling application *25HoursADay*. This developer guide covers an overview of the program architecture, the logical view of major components, and the mechanism of the functions.
+
+The intended audience of this guide are the developers and software testers of *25HoursADay*.
+
+### Developer guide organization
+The table below shows the content of the developer guide.
+
+|Section| Purpose | 
+|--------|----------|
+|[Section 2: Setting up](#setting-up)|To provide instructions on how to download and set up *25HoursADay* on your computer.|
+|[Section 3: Design](#design)|To explain the architecture of *25HoursADay*,  and describe the major components of *25HoursADay*, the roles of major components, as well as their organization and interaction of major components.|
+|[Section 4: Implementation](#implementation)|To explain the mechanism of functions added to *25HoursADay* and the features added to *25HoursADay*.|
+|[Section 5: Documentation](#documentation)|To represent documents describing the system and its parts.|
+
+## Setting Up & Getting started
+This section provides on setting up the *25HoursADay* application.
+
+### Setting up
+
+Before diving into the project development and testing, here are some essential tools required in this project:
+
+1. **JAVA 11.0.8**
+    * Launch your terminal and type "java -version" to ensure you have the correct version number
+    * If you have no Java installed or different version number, please proceed to this [website](https://www.oracle.com/java/technologies/javase/jdk11-archive-downloads.html)
+    to download the **Java 11.0.9** installer.
+1. **Intellij IDEA**
+    * If you are new to IntelliJ, please proceed to this [website](https://www.jetbrains.com/help/idea/installation-guide.html)
+    to download and install the IDE.
+    
+After installing the required tools, proceed to [Github Page](https://github.com/AY2021S1-CS2113T-T12-2/tp) and 
+fork the project to your own repository, and clone your fork into your working computer.    
+
+Now, Open Intellij 
+1. If you are not in the welcome screen, click `File` → `Close Project` to navigate to the starting project dialog)
+1. Set up the correct JDK version for Gradle.
+        1. Click `Configure` → `Project Structure for New Project`
+        1. Click `New...` and set it to the directory of the JDK
+1. Upon the successful configuration, you should be able to see the following screenshot:
+![JDK Config](images/JDK_Config_Screenshot.png)
```

Comment:
I like how you insert the screenshot of the configuration here so that the readers can confirm that they configure the ide correctly.